### DISCUSSION TITLE
Is it possible to use an existing meshcentral serveur with tactical rmm view

### DISCUSSION THREAD

**Discussion by Scaff31** [amidaware/tacticalrmm#384] [Q&A]
Hello all and thx for your work.

I've m'y own meshcentral serveur with 50groups ans 300agents (using CIRA toi) already deploy

Is it possible to use the tactical view with my own MC server?

Thx for answer
Scaff

**Comment by dinger1986** [comment]
Yes (copied from a previous ticket) from @wh1te909 

Hi, so when you install tactical, it also installs meshcentral for you with a config.json that allows integration via iframes and login token.

You don't have to use the meshcentral that tactical installs. You can use your existing one and just point tactical to use that.

I need to write up some docs on how to do that but basically just look at the config.json that tactical installs and make sure your existing mesh's config has "AllowLoginToken": true and "AllowFraming": true

Then you'll need to generate a login token on your existing mesh server:
(just change the path to to wherever your meshcentral is installed)

node /meshcentral/node_modules/meshcentral --logintokenkey
If you already have a meshcentral user account that has access to all device groups then you can use that, if not, create a new mesh user with full access to all mesh agents

Then from Tactical's web UI, go to Settings > Global Settings > MeshCentral

Username: the meshcentral username that has access to all device groups/mesh agents
Mesh Site: https://mesh.yourdomain.com
Mesh Token: copy and paste the token from the command above

https://wh1te909.github.io/tacticalrmm/mesh_integration/

@wh1te909 maybe worth adding the above to the docs?

**Reply by Scaff31** [reply]
Thx for your gast Replay

I see the integration part on Doc it's more understandable with your explanation.

Sorry for my Bad english it's hard on phone with auto correct on another language

**Reply by dinger1986** [reply]
No problems, hope that helps as I haven't actually ever done the integration myself so that's as much as I can help with. Plenty of people on discord who would help I am sure.

**Reply by Scaff31** [reply]
Another question,

All of my service was behind HAproxy can i setup tactical behind HAproxy with TLSOffloading  ?

**Reply by dinger1986** [reply]
Pretty sure you can and others have.

Jump on the discord channel there might be someone about there who can help with that.

**Reply by bradhawkins85** [reply]
I have HAProxy in front of my TRMM doing TLS termination and it works but has a few bugs.
Script timeouts, the web UI doesn't wait more than 75 seconds for a reply from the script, the script still runs but returns and empty result.
Mesh central used to work on older versions but now stops communicating due to SSL errors as the client is expecting a different certificate to the one coming from HAProxy. I have changed the port to 4443 for Mesh client comms and forwarded it directly to the rmm to get around this.
I have posted a config in another now closed issue with HAProxy config, check the closed issues it's not too hard to find.
